### ISSUE TITLE
Design a more coherent index API

### ISSUE THREAD

**Issue by wjones127** [lancedb/lancedb#1666]
There's some design work we need to do on indices. We don't represent them in our APIs very coherently. There's sort of three levels we can imagine:

1. A configuration for an index ("This is an IVF_PQ index")
2. An index on a column ("These are the index files for an IVF_PQ index")
3. A piece / segment of an index ("This is one index file for an IVF_PQ index covering fragments ...")

In most databases, users kind of expect they are creating 1. That's why they can do (1) create table, (2) create index, then (3) insert data. They are confused when the need to insert data before creating an index.

What we have right now is 3. This relates to the whole UUID confusion we had earlier, where not only does the UUID change as you incrementally update the index. But also you can have multiple UUIDs for the same index name! That's because in Lance an "index" really refers to a piece of an index.

**Comment by LuQQiu** [comment]
So list_indices with same name are showing the **Index files, unique by UUID** instead  of top-level indices?

**Comment by wjones127** [comment]
> So list_indices with same name are showing the **Index files, unique by UUID** instead of top-level indices?

yes, list indices shows files. There isn't a proper top-level index concept in Lance yet. So the "duplicates" you see are just different pieces of an index.

**Comment by rpgreen** [comment]
We may also want to consider exposing index status and async jobs as part of this model.
